### ISSUE TITLE
New submodules aren't initialized on checkout

### ISSUE THREAD

**Issue by mtschoen-unity** [ForkIssues/Tracker#620]
I'm not sure if this is by design, but I have found that if I am checking out a commit that adds submodules to a repository, fork will not initialize these, while it does update others. Normal behavior for a checkout is not to update any submodules, but since Fork is doing me a favor and updating existing submodules, I figure throwing `--init` into that command couldn't hurt?

**Comment by DanPristupov** [comment]
So, 'Update submodules automatically' is enabled in preferences in your case, right?

> I figure throwing --init into that command couldn't hurt

I will check this, but usually adding more options to checkout operation usually slows down the performance on large repos.

**Comment by DanPristupov** [comment]
I just checked the source code. It does include `--init`:

```git submodule update --init --recursive```

Please check the 'Update submodules automatically' option.

**Comment by mtschoen-unity** [comment]
Hm! I wasn't aware of that option. I'll try again later with that checked to confirm. But then, if I didn't have it checked, why would I be getting submodule updates that _aren't_ inits? I may just be mistaken about whether or not the checkout should be updating submods, so I'll close this until I run into trouble again.

As always, thanks for the prompt reply!

**Comment by DanPristupov** [comment]
> why would I be getting submodule updates that aren't inits?

This should not be true, please double check this.

**Comment by mtschoen-unity** [comment]
I had it happen again. I double-checked that "update submodules automatically" was checked, and after checking out a new remote branch, the submodules show up in Fork but there's nothing in the folder. I have to go to the terminal and run the init.

**Comment by DanPristupov** [comment]
I reproduced the problem and fixed it. The fix will appear in 1.0.81.

**Comment by mtschoen-unity** [comment]
Thanks! You da man 💪

**Comment by DanPristupov** [comment]
Fork 1.0.81 is released.

**Comment by mtschoen-unity** [comment]
Hm. I actually just tried this again and still get an empty folder for new submodules. Maybe it's because I also have some submodules checked out already? It looks like the submodule update didn't run/complete because the other submodules show up as changes after this particular checkout.

**Comment by DanPristupov** [comment]
That's strange. I've spent much time testing that new submodules get downloaded on checkout.

OK, I just tested:

* Ensure 'Update submodules automatically' is enabled
* create empty repo, add initial commit (1).
* add submodule and commit that (2).
* checkout the initial commit (2)
* open finder and remove /repo/submodule and /repo/.git/modules. Clear Trash to be sure :).
* checkout the (2) commit. It takes some time, then /repo/submodule and /repo/.git/modules appears again

Did I miss something?
